doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -1,394 +1,648 @@
+    target_pose.orientation.x = 0.000;
+    target_pose.orientation.y = 0.000;
+    target_pose.orientation.z = 0.707;
+    target_pose.position.x = 0.5;
+    target_pose.position.y = -0.2;
+    target_pose.position.z = 0.5;
+    move_group.setPoseTarget(target_pose);
+
+    // Plan
+    bool success = (move_group.plan(my_plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
+    ROS_INFO_NAMED("tutorial", "Visualizing plan 6 (pose goal move around cuboid) %s", success ? "" : "FAILED");
+
+    // Visualize the plan in RViz'
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishText(text_pose, "Pose Goal Avoiding Obstacle", rvt::WHITE, rvt::XLARGE);
+    visual_tools.publishAxisLabeled(target_pose, "point");
+    visual_tools.publishTrajectoryLine(my_plan.trajectory_, joint_model_group);
+    visual_tools.trigger();
+  }
+
+  void remove_obstacle()
+  {
+    ROS_INFO_NAMED("tutorial", "Remove the object from the world");
+
+    // BEGIN_SUB_TUTORIAL remove_obj
+    //
+    // Now, let's remove the collision object from the world.
+    planning_scene_interface.removeCollisionObjects(object_ids);
+    object_ids.pop_back();
+    collision_objects.pop_back();
+    // **Note:** Objects in the world doesn't change how to plan or execute that plan.
+    //
+    // END_SUB_TUTORIAL
+
+    // RViz
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishText(text_pose, "Object Removed", rvt::WHITE, rvt::XLARGE);
+    visual_tools.trigger();
+    visual_tools.prompt("Press 'next' to create a new object");
+  }
+
+  void create_parcel()
+  {
+    // Define a collision object ROS message
+    moveit_msgs::CollisionObject collision_object;
+    collision_object.header.frame_id = move_group.getPlanningFrame();
+
+    // The id of the object is used to identify it
+    collision_object.id = "box2";
+
+    // Define a box to add to the world
+    shape_msgs::SolidPrimitive primitive;
+    primitive.type = primitive.BOX;
+    primitive.dimensions.resize(3);
+    primitive.dimensions[0] = 0.1;
+    primitive.dimensions[1] = 0.1;
+    primitive.dimensions[2] = 0.1;
+
+    // Define a pose for the box (specified relative to frame_id)
+    geometry_msgs::Pose box_pose;
+    box_pose.orientation.w = 1.0;
+    box_pose.position.x = 0.5;
+    box_pose.position.y = -0.2;
+    box_pose.position.z = 0.6;
+
+    collision_object.primitives.push_back(primitive);
+    collision_object.primitive_poses.push_back(box_pose);
+    collision_object.operation = collision_object.ADD;
+
+    collision_objects.push_back(collision_object);
+    object_ids.push_back(collision_object.id);
+
+    // Add the collision object into the world
+    ROS_INFO_NAMED("tutorial", "Adding an object into the world");
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishText(text_pose, "Attach Object", rvt::WHITE, rvt::XLARGE);
+    visual_tools.trigger();
+
+    planning_scene_interface.addCollisionObjects(collision_objects);
+  }
+
+  void attach_parcel()
```

Comment:
attach what parcel? Treat this as an api that someone else might want to use for their own use case and make these methods re-usable